src/common.jl

```diff
@@ -298,6 +300,47 @@ function Base.iszero(p::AbstractPolynomial)
     return all(iszero.(coeffs(p))) && p[0] == 0
 end
 
+# See discussions in https://github.com/JuliaMath/Polynomials.jl/issues/258
+is_(fn, p::AbstractPolynomial) = all(fn, coeffs(p))  # Do not export
+as_(fn, p::P) where {P<:AbstractPolynomial} = ⟒(P)(fn.(coeffs(p)), p.var)  # Do not export
+
+"""
+    isreal(p::AbstractPolynomial)
+
+Determine whether a polynomial is a real polynomial, i.e., having only real numbers as coefficients.
+
+See also: [`real`](@ref)
+"""
+Base.isreal(p::AbstractPolynomial) = is_(isreal, p)
+"""
+    real(p::AbstractPolynomial)
+
+Construct a real polynomial from the real parts of the coefficients of `p`.
+
+See also: [`isreal`](@ref)
+
+!!! note
+    This could cause losing terms in `p`. This method is usually called on polynomials like `p = Polynomial([1, 2 + 0im, 3.0, 4.0 + 0.0im])` where you want to chop the imaginary parts of the coefficients of `p`.
+"""
+Base.real(p::AbstractPolynomial) = as_(real, p)
+
+"""
+    isintegerpoly(p::AbstractPolynomial)
+
+Determine whether a polynomial is an integer polynomial, i.e., having only integers as coefficients.
+
+See also: [`asintegerpoly`](@ref)
+"""
+isintegerpoly(p::AbstractPolynomial) = is_(isinteger, p)
+"""
+    asintegerpoly(p::AbstractPolynomial)
+
+Construct an integer polynomial from the coefficients of `p`.
+
+See also: [`isintegerpoly`](@ref)
+"""
+asintegerpoly(p::AbstractPolynomial) = as_(x -> convert(Integer, x), p)
```

Comment:
My only hesitation is the name of these two, especially `asintegerpoly`. Is this an improvement over `convert(Polynomial{Int}, p)` (an alternate way to do this particular task, that might be more intuitive)? As well, would a user expect that `isinteger(p)` would work before hunting for a `isintegerpoly(p)` call? I know `isinteger` isn't extended to vectors, but I'm not so sure what is preferable slightly bending the generic usage or coming up with a new name.

I'm okay, personally, with one of these, but if there are more desired, we might need to reconsider.

Reply:
I don't like `asintegerpoly` either, maybe we can deprecate it and add a doc to `isintegerpoly`? Let me ask the Julia developers whether `isinteger` can be applied onto a `Vector`.

Reply:
Yeah, that makes sense. Not sure how `isreal` survived through that tightening up. Let's say `isintegral`.